athena-jdbc/src/main/java/com/amazonaws/athena/connectors/jdbc/connection/DatabaseConnectionConfigBuilder.java

```diff
@@ -144,8 +144,9 @@ private DatabaseConnectionConfig extractDatabaseConnectionConfig(final String ca
     private Optional<String> extractSecretName(final String jdbcConnectionString)
     {
         Matcher secretMatcher = SECRET_PATTERN.matcher(jdbcConnectionString);
+        boolean isValidGroupCount = secretMatcher.groupCount() == 1 || secretMatcher.groupCount() == 2;
```

Comment:
why is secretMatcher.groupCount() == 2 valid? doesn't look like were doing anything with the 2nd item

Reply:
Also we currently don't use the Service ID, which is the 2nd item, but it may be helpful to have in the future if the service linked secrets become more wide spread.

Reply:
do we actually care about the serviceid? I think thats a semantic for secrets manager.

Reply:
shouldn't this work?

```
\\$\\{([a-zA-Z0-9:!/_+=.@-]+)}
```

Reply:
nvm we don't want this, we should restrict the first part before the ! character since thats specific to service managed secrets